v3/routes/exercises.rb

```diff
@@ -18,6 +20,14 @@ class Exercises < Core
         status 404 unless problem.exists?
         pg :"exercise/exists", locals: { problem: problem }
       end
+
+      get '/tracks/:id/exercises/:slug' do |id, slug|
+        problem = config.find(id).find(slug)
+        file = Tempfile.new('data.zip')
+        ZipFileGenerator.write(problem.send(:dir), file.path)
```

Comment:
I think `dir` is a private method of `problem` https://github.com/exercism/x-api/blob/master/lib/xapi/problem.rb#L73